src/main/java/com/ypeckstadt/service/account/AccountServiceForTransaction.java

```diff
@@ -0,0 +1,107 @@
+package com.ypeckstadt.service.account;
+
+import com.google.protobuf.ServiceException;
+import com.scalar.db.api.DistributedTransaction;
+import com.scalar.db.api.DistributedTransactionManager;
+import com.scalar.db.exception.transaction.CommitException;
+import com.scalar.db.exception.transaction.UnknownTransactionStatusException;
+import com.ypeckstadt.dao.DaoException;
+import com.ypeckstadt.dao.ScalarDbManager;
+import com.ypeckstadt.dao.account.AccountDao;
+import com.ypeckstadt.dao.account.AccountRecord;
+
+public class AccountServiceForTransaction extends AccountService {
+
+    private final DistributedTransactionManager transactionManager;
+
+    public AccountServiceForTransaction(AccountDao accountDao, ScalarDbManager scalarDbManager) {
+        super(accountDao);
+        this.transactionManager = scalarDbManager.getDistributedTransactionManager();
+    }
+
+    @Override
+    public AccountRecord create(String id) throws Exception {
+
+        // start the transaction
+        DistributedTransaction transaction = transactionManager.start();
+
+        try {
+            // Check if account exists
+            if (accountDao.exists(id, transaction)) {
+                transaction.abort();
+                throw new Exception("the account already exists");
+            }
+
+            // Save new account
+            accountDao.put(id, 0, System.currentTimeMillis(), transaction);
```

Comment:
It's a good practice (not documented yet) to catch CrudConflictException thrown by `put()` and CommitConflictException thrown by `commit()` and retry the transaction for better use experience.
Both exceptions are thrown due to conflicting transactions, so there are a lot of chances the transaction can go through when retried.

Reply:
I will update this today. We have noticed that the CrudConflictException is only available in the master branch though. I guess it is a new type of exception.